src/tests/tracing/runtimeeventsource/NativeRuntimeEventSourceTest.cs

```diff
@@ -31,14 +32,19 @@ static int Main(string[] args)
                     if (OperatingSystem.IsWindows())
                         DoOverlappedIO();
 
-                    // Wait for events.
-                    Thread.Sleep(1000);
-
                     // Generate some GC events.
                     GC.Collect(2, GCCollectionMode.Forced);
 
-                    // Wait for more events.
-                    Thread.Sleep(1000);
+                    Stopwatch sw = Stopwatch.StartNew();
+
+                    while (sw.Elapsed <= TimeSpan.FromMinutes(1))
+                    {
+                        Thread.Sleep(100);
+                        if (listener.SeenProvidersAndEvents.Contains("Microsoft-Windows-DotNETRuntime/EVENTID(65)"))
```

Comment:
Might also want to guard these with lock to avoid race conditions. Contains can throw NRE on concurrent access and the foreach below IOE.

Reply:
Thanks, switched to ConcurrentBag for ease of implementation.